src/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -108,7 +74,6 @@ public sealed partial class JsonBoolean : System.Text.Json.JsonNode, System.IEqu
         public bool Equals(System.Text.Json.JsonBoolean other) { throw null; }
         public override int GetHashCode() { throw null; }
         public static bool operator ==(System.Text.Json.JsonBoolean left, System.Text.Json.JsonBoolean right) { throw null; }
-        public static implicit operator System.Text.Json.JsonBoolean (bool value) { throw null; }
```

Comment:
Is there a downside of keeping this as well? It will be a little weird to have:
```c#
JsonNode aBoolean = false; // Works just fine, and aBoolean is a JsonBoolean
JsonBoolean anotherBoolean = true; // Fails because we don't have implicit converters.
```

Reply:
Then probably I would need to add implicit conversions for JsonNumber as well and there is this problematic issue of throwing for values like float.PositiveInfinity (currently I am creating JsonString in such case for JsonNode implicit cast operator).

Reply:
We should document that reasoning for why we don't have the implicit casts on the concrete types in the design doc. Something worth re-visiting in the future.